### PR TITLE
Itkdev: interest period config

### PR DESCRIPTION
**Pull request by sinejespersen**
#### Link to issue

n/a

#### Description

Change format of interest periods

https://github.com/danskernesdigitalebibliotek/dpl-react/pull/463

#### Screenshot of the result

image

#### Checklist

- [x] My complies with [our coding guidelines](../docs/code_guidelines.md).
- [x] My code passes our static analysis suite. If not then I have added a comment explaining why this change should be exempt from the code standards and process.
- [x] My code passes our continuous integration process. If not then I have added a comment explaining why this change should be exempt from the code standards and process.

#### Additional comments or questions

n/a

### CODE REVIEW

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php [hunk 1]
```diff
@@ -155,6 +155,16 @@ public function search(): array {
...
   }
 
+  /**
+   * Get a string with interest periods.
+   *
+   * @return string
+   *   A string with interest periods
+   */
+  public static function getInterestPeriods(): string {
+    return \Drupal::configFactory()->get('dpl_library_agency.general_settings')->get('interest_periods_config');
```
Comment (spaceo) [inline web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:165]: I think you should add the general settings as a dependency like the other settings:
https://github.com/danskernesdigitalebibliotek/dpl-cms/blob/0c6d60916b6425243ef1c9b0ce3871b2eab9c4ee/web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php#L44-L48
To keep it persistent and make it easier to test.

**Review by spaceo — CHANGES_REQUESTED** [review]
Looks good. But please look at my comment.

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:165]: That is not possible here as the function is static and don't have access to `$this`.

**Review by cableman — APPROVED** [review]

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php (hunk 1 above)
Comment (spaceo) [reply web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:165]: Ok I have been digging into the various ways we are handling settings and it is not consistent.
I understand why you need to solve it like this, but I think we can make it more consistent by creating services for the different areas of the configuration (like `DplInstantLoanSettings`) in the future.

**Review by spaceo — APPROVED** [review]
Approved but please look at my comment

File: web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php (hunk 1 above)
Comment (cableman) [reply web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:165]: I hear you but I don't think that this PR should change how the general settings is handled. But "yes" to the idea to change this into the settings pattern now applied in may modules.

Comment (cableman) [reply web/modules/custom/dpl_react_apps/src/Controller/DplReactAppsController.php:168]: Todo added and ticket on the way about fixing this into the settings pattern

**Pull request merged**